lib/puppet/util/windows/user.rb

```diff
@@ -22,6 +22,14 @@ def admin?
   # https://msdn.microsoft.com/en-us/library/windows/desktop/ee207397(v=vs.85).aspx
   SECURITY_MAX_SID_SIZE = 68
 
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/ms681385(v=vs.85).aspx
+  # These error codes indicate successful authentication but failure to
+  # logon for a separate reason
+  ERROR_ACCOUNT_RESTRICTION = 1327
+  ERROR_INVALID_LOGON_HOURS = 1328
+  ERROR_INVALID_WORKSTATION = 1329
+  ERROR_ACCOUNT_DISABLED    = 1331
```

Comment:
There was another code I considered, `ERROR_PASSWORD_EXPIRED = 1330` but since the password setter also sets the password to not expire I figured there wasn't much value in it? I suppose it could avoid a password set on the very first run.